Add cleanup functionality for extracted ZIP files

ammaarkhan opened the pull request:
<!-- 
Thank you for contributing! Please fill out this template to help us review your PR.
-->

## 📝 Description

- Deletes the zip_data/ folder after parsing and analysis is complete.

**Closes:** #119

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation added/updated
- [ ] ✅ Test added/updated
- [x] ♻️ Refactoring
- [x] ⚡ Performance improvement

---

## 🧪 Testing

### Manual verification (zip_data cleanup)
1. Activate the virtualenv: `source venv/bin/activate`.
2. Confirm baseline state: run `ls zip_data` (expect “No such file or directory”).
3. Launch the CLI with `python src/main.py`, accept both consent prompts, provide the path to a sample ZIP that contains at least one project, and complete the classification/analysis prompts.
4. While analysis runs, run `ls zip_data` to confirm the extracted folder is present.
5. After the CLI exits, run `ls zip_data` again and verify the directory no longer exists (or is empty if the OS kept the folder). Capture the cleanup log line if possible.

### Automated
- [ ] `pytest`

---

## ✓ Checklist

- [x] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [x] 💬 I have commented my code where needed
- [ ] 📖 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [ ] 🔗 Any dependent changes have been merged and published in downstream modules
- [ ] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

> If applicable, add screenshots to help explain your changes

<details>
<summary>Click to expand screenshots</summary>

<!-- Add your screenshots here -->

</details>

ammaarkhan committed:
Add cleanup functionality for extracted ZIP files

taoTimTim reviewed (approved):
Good work, this does exactly what we need it to (which is to delete the folder after parsing).

A couple of notes:
- Don't delete `zip_data` inside `prompt_and_store()`. Although that function does contain the majority of the flow, it mostly deals with parsing and data routing. It is already quite large and will likely need to be refactored in the future, so it's best to separate deleting `zip_data` from it for maintainability/future implementations. Instead, I would make the function `prompt_and_store()` return the last ZIP path (the path to the folder `zip_data`), then call `cleanup_extracted_zip()` inside main. Thus, any future changes to the parsing will not need to worry about timing.

**Proposed Solution:**
```
def main() {
   ...
   processed_zip_path = prompt_and_store()
   cleanup_extracted_zip(processed_zip_path)
}
```
```
def prompt_and_store() {
   ...
   return processed_zip_path
}
```

- Please implement unit tests for this, to ensure better test coverage for the system as a whole. You will probably need to create a temporary test folder, call `cleanup_extracted_zip()` on that folder, and make sure it no longer exists.

If you think deleting `zip_data` inside `prompt_and_store()` is still the best approach that's totally fine, the above suggestion is mainly to help keep responsibilities separate and avoid a massive control flow cycle, something we have slowly been refactoring to get rid of the past week.

The code itself it great, just the flow needs some change. Thanks for working on this, Ammaar!

taoTimTim reviewed (changes requested):
Accidentally approved my last review, adding a second to request changes. Sorry!

ammaarkhan committed:
Implement cleanup of extracted ZIP files in main flow and add unit tests for cleanup functionality

ammaarkhan commented:
@taoTimTim thanks for the feedback! Completely agree with your suggestion. The initial was approach was a bit messy. Updated it as requested and added a unit test.

taoTimTim reviewed (approved):
All tests are passing and the change of moving the deletion from `prompt_and_store()` to `main()` is working great. Approved.

johaneshp reviewed (approved):
all tests passed. functions work as described. good work ammaar!
